libnmstate/schema.py

```diff
@@ -0,0 +1,31 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+#
+# Refer to the README and COPYING files for full details of the license
+#
+from __future__ import absolute_import
+
+import json
+import pkgutil
+
+
+def load(schema_name):
```

Comment:
Why do we want to load sub-schemas? What would be a scenario for that? I imagine we get data with full description on input, we validate against the whole schema and that's it. Is it here because of we want to allow plugins to register for specific sub-schema?

Reply:
I do not have context here to your comment. We load a schema (or a list of them in the future, not sure yet), and can validate against it.
With the json-schema, one schema can reference others.

Reply:
In my opinion we should have main schema with the object that references other types (defined in other schema files). These other schema files will not contain the top level object, only definitions, therefore they cannot be used stand-alone for validation.

My question is, do we have a use case for validating subtrees? If not, we can use other schema files only for definitions.

Reply:
This is the current state, unless we find a different need, I guess we will stick with it.
The interfaces schema is currently the root schema, we may change it when the routing subtree will be introduced.

Reply:
I see now, thanks. All good then.